mlflow/langchain/__init__.py

```diff
@@ -271,6 +271,8 @@ def load_retriever(persist_directory):
     _validate_and_prepare_target_save_path(path)
 
     if isinstance(model_config, str):
```

Comment:
We might need to add this check here: https://github.com/mlflow/mlflow/blob/master/mlflow/utils/model_utils.py#L383 as well for pyfunc to work.

also can we reuse this util here? 
Replace this entire code with _validate_and_get_model_config_from_file?

Reply:
+1